src/compiler/checker.ts

```diff
@@ -9398,13 +9944,21 @@ namespace ts {
         }
 
         function getReturnTypeFromBody(func: FunctionLikeDeclaration, contextualMapper?: TypeMapper): Type {
+            let type: Type;
+            if (func.parserContextFlags & ParserContextFlags.JavaScriptFile) {
+                type = getReturnTypeFromJSDocComment(func);

```

Comment:
I'm surprised we do this here vs. in `getSignatureFromDeclaration`. Any reason why?